src/coreComponents/constitutive/solid/Damage.hpp

```diff
@@ -132,21 +143,88 @@ class DamageUpdates : public UPDATE_BASE
 
   inline
   GEOS_HOST_DEVICE
-  virtual real64 getDegradationDerivative( real64 const d ) const
+  virtual real64 getDegradationDerivative( localIndex const k, real64 const d ) const
   {
+    GEOS_UNUSED_VAR( k );
+
     return -2*(1 - d);
   }
 
 
   inline
   GEOS_HOST_DEVICE
-  virtual real64 getDegradationSecondDerivative( real64 const d ) const
+  virtual real64 getDegradationSecondDerivative( localIndex const k, real64 const d ) const
   {
-    GEOS_UNUSED_VAR( d );
+    GEOS_UNUSED_VAR( k, d );
 
     return 2.0;
   }
 
+  //Damage dependence function on fluid pressure terms and its derivatives
+
+  GEOS_FORCE_INLINE
+  GEOS_HOST_DEVICE
+  virtual real64 pressureDamageFunction( localIndex const k,
+                                         localIndex const q ) const
+  {
+    real64 pf = fmax( fmin( 1.0, m_newDamage( k, q )), 0.0 );
+
+    return 0.5*(1 + std::cos( M_PI*pf ));
+  }
+
+
+  GEOS_FORCE_INLINE
+  GEOS_HOST_DEVICE
+  virtual real64 pressureDamageFunctionDerivative( real64 const d ) const
+  {
+    return -0.5*M_PI*std::sin( M_PI*d );
+  }
+
+
+  GEOS_FORCE_INLINE
+  GEOS_HOST_DEVICE
+  virtual real64 pressureDamageFunctionSecondDerivative( real64 const d ) const
+  {
+    return -0.5*M_PI*M_PI*std::cos( M_PI*d );
+  }
+
+  GEOS_FORCE_INLINE
+  GEOS_HOST_DEVICE
+  virtual real64 getDamage( localIndex const k,
+                            localIndex const q ) const
+  {
+    return m_newDamage( k, q );
+  }
+
+  GEOS_FORCE_INLINE
+  GEOS_HOST_DEVICE
+  virtual real64 getOldDamage( localIndex const k,
+                               localIndex const q ) const
+  {
+    return m_oldDamage( k, q );
+  }
+
+  GEOS_FORCE_INLINE
+  GEOS_HOST_DEVICE
+  virtual void getDamageGrad( localIndex const k,
+                              localIndex const q,
+                              real64 ( & damageGrad )[3] ) const
+  {
+    for( int dim=0; dim < 3; ++dim )
+    {
+      damageGrad[dim] = m_damageGrad[k][q][dim];
+    }
+
+  }
+
+  GEOS_FORCE_INLINE
+  GEOS_HOST_DEVICE
+  virtual void updateBiotCoefficient( localIndex const k,
+                                      real64 const biotCoefficient ) const
+  {
+    m_biotCoefficient[k] = biotCoefficient;
```

Comment:
it's a bit weird to have the biotCoefficient here. Is it really needed? Can't it be done by `PorousDamageSolid`?

Reply:
It will be needed in [PhaseFieldPressurizedDamageFEMKernels](https://github.com/GEOS-DEV/GEOS/blob/9b7b02faaec1c1e8242d86f3eb78b4ec84311d29/src/coreComponents/physicsSolvers/simplePDE/PhaseFieldPressurizedDamageFEMKernels.hpp), which currently only uses the `SolidModel` type instead of `CoupledSolid`. We can change the constitutive model type in `PhaseFieldPressurizedDamageFEMKernels.hpp` to `PorousDamageSolid`